### ISSUE TITLE
Support Xdebug?

### ISSUE THREAD

**Issue by SalvatorePollaci** [ezsystems/launchpad#39]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | no
| Version          | x.y.z
| Environment      | Linux/Mac/Windows

It would be very nice if eZ Launchpad supported xdebug.

Could this be implemented?

**Comment by Plopix** [comment]
@yhanini can you do a PR, also I would like more

To me `xdebug` can reduce performances, then I would love to be able to enable/disable it very easily. (using the `provisionning/engine/php.ini` + `ez up`
=> today in your code installation is triggered by the env var but installation could be in the image actually

- [ ] action 1: doc a PR on https://github.com/Plopix/docker-php-ez-engine to add `xdebug` but disabled by default

XDEBUG_REMOTE_AUTOSTART and XDEBUG_REMOTE_CONNECT_BACK could be hardcode right? not sure if we should let then in the docker-compose file. same with XDEBUG_REMOTE_HOST is that the "host" then I think that could be automated.

=> actually I think all those value should be put in the `php.ini` we don't even need a script
Those are just xdebug properties

- [ ] action 2: add the correct configuration (commented out) in the`php.ini`

To sum up:

At then end we should have the xdebug properties all setup in the `provisionning/engine/php.ini` (with xdebug disabled by default)

And to enable it, the procedure should be to just enable it in this `provisionning/engine/php.ini` and run `ez up`.

=> that is it!

- [ ] action 3: add the documentation

**Comment by kemoc** [comment]
I ask to add more settings configurable by ENVs, my example xdebug settings without docker:
`[xdebug]
xdebug.max_nesting_level = 500
xdebug.remote_autostart = 0
xdebug.remote_enable = 1
xdebug.remote_host = localhost
xdebug.remote_port = 9000
xdebug.idekey = "PHPStorm"
xdebug.profiler_enable_trigger = 1
; default cachegrind.out.%p
xdebug.profiler_output_name = cachegrind.out.%s
xdebug.profiler_output_dir = "/tmp/xdebug/profiler/output"
xdebug.trace_output_dir = "/tmp/xdebug/trace/output"
xdebug.remote_cookie_expire_time = 7200`

so for me important are enable/disable Xdebug by Cookie or GET, for it will be needed in container
XDEBUG_IDEKEY
XDEBUG_REMOTE_ENABLE
but xdebug.max_nesting_level should be configurable too, so
XDEBUG_MAX_NESTING_LEVEL

**Comment by kemoc** [comment]
@Plopix
> XDEBUG_REMOTE_AUTOSTART and XDEBUG_REMOTE_CONNECT_BACK could be hardcode right? 
I did not understand what you mean hardcoded but for me XDEBUG_REMOTE_AUTOSTART must able to set value by .env and/or docker-compose.yml

**Comment by yhanini** [comment]
PR : https://github.com/ezsystems/launchpad/pull/43

**Comment by Plopix** [comment]
https://github.com/ezsystems/launchpad/pull/44/commits/857d991f1881b3bbe4a1163d367e858240eb57da

**Comment by Plopix** [comment]
enjoy ! https://github.com/ezsystems/launchpad/pull/44

**Comment by Plopix** [comment]
@kemoc you can now do all the config that you want  through the `php.ini`
